### DISCUSSION TITLE
Live-view / camera communication problem

### DISCUSSION THREAD

**Discussion by Karl0202** [photobooth-app/photobooth-app#383] [General]
Hello!

I'm currently testing the Photobooth-app with my hardware.

- Raspi 5, 4GB, 128GB SSD
- Canon M50
- Cp1500
- App Version 4.1.1
- 
Everything works reliably and error-free for hours. However, sometimes the communication to the camera seems to break off, usually the live stream no longer works, but photos can still be taken. Sometimes, however, no more pictures can be taken. However, it is often enough to trigger a video to re-establish communication. Sometimes the software needs to be restarted. The camera always remains switched on.
I also had a lot of messages in my log files: ‘(ptp_usb_getresp [usb.c:516]) PTP_OC 0x9153 receiving resp failed: Camera Not Ready (0xa102) (port_log.py:20)’. So in the file gphoto2.py I have the inserted pause of time.sleep(0.02) to time.sleep(0.04). The messages have disappeared since then.
In the attached log file, the first errors can apparently be seen from line 2929. Can one of the experts recognise where I can start to fix the problem?

Thanks for your help!
[photobooth_20241128.log](https://github.com/user-attachments/files/18227506/photobooth_20241128.log)

**Comment by mgineer85** [comment]
Hi, 

thanks for your report. 
I found Pi5 also being too fast for the DSLR gphoto2 backend and added a sleep earlier: https://github.com/photobooth-app/photobooth-app/blob/d770e0d0b8e6db55362e9562258dc4cb1a84669f/photobooth/services/backends/gphoto2.py#L299

For my old Canon 1100D it eliminated the Camera not ready issue completely - maybe different cameras need different delays?  At which position did you add the sleep?

```
2024-11-28 22:37:58,973 [    INFO] (gp_context_error) Canon EOS M Capture failed: Perhaps no focus? (port_log.py:20)
```
Can you confirm the camera can focus properly and has internal standby disabled?
What if you switch to manual focus, would that improve situation?

**Reply by Karl0202** [reply]
I have only enlarged the ‘sleep’ you inserted. However, this has not affected the general misbehaviour. Only the logs remain clean (so far).
I can still test the manual focus, but I would almost rule it out as the main problem, as the communication is then completely stuck (no more live view) and doesn't work with the next image either. Standby is also off.
What do you think, what else can I try to localise the error?

**Reply by mgineer85** [reply]
It seems there is a slightly newer [gphoto libgphoto2 available](https://github.com/gphoto/libgphoto2/releases) and according to the changelogs it has some bugfixes for Canon cameras:

`pipx install --system-site-packages photobooth-app --pip-args='--prefer-binary'`

`pipx install --system-site-packages photobooth-app --pip-args='--prefer-binary --only-binary gphoto2'` (untested)

When the updated version including the binaries is applied, you would read libgphoto2 to be version 2.5.31

```
2024-11-28 20:09:52,947 [    INFO] started WledService (container.py:86)
2024-11-28 20:09:52,968 [    INFO] python-gphoto2: 2.5.1 (gphoto2.py:78)
2024-11-28 20:09:52,968 [    INFO] libgphoto2: ['2.5.30', 'standard camlib set (ax203 canon digigr8 dimagev directory docupen jl2005a jl2005c kodak_dc240 lumix mars pentax ptp2 ricoh_g3 sierra sonix sq905 st2205 topfield tp6801)', 'gcc (C compiler used)', 'no ltdl (for portable loading of camlibs)', 'EXIF (for special handling of EXIF files)'] (gphoto2.py:79)
2024-11-28 20:09:52,968 [    INFO] libgphoto2_port: ['0.12.1', 'iolibs: disk ptpip serial usb1 usbdiskdirect usbscsi', 'gcc (C compiler used)', 'no ltdl (for portable loading of iolibs)', 'EXIF (for vusb)', 'USB (libusb1, for USB cameras)', 'serial (for serial cameras)', 'no resmgr (serial port access and locking)', 'no ttylock (serial port locking)', 'no lockdev (serial port locking)'] (gphoto2.py:80)
2024-11-28 20:09:52,968 [    INFO] aquisition main backend: <class 'photobooth.services.backends.gphoto2.Gphoto2Backend'> (aquisitionservice.py:63)
```


Also a web research for gphoto2 in combination with your camera would be a good idea.

**Reply by mgineer85** [reply]
It seems, that the error is not correctly catched and thus a restart of the backend fails. I will look into this also.

**Reply by Karl0202** [reply]
Hi!
Unfortunately I am not able to upgrade the libgphoto2 version as you described. The commands run without errors, but in the log file the version remains at 2.5.30. How could I install the new version? Do you have any ideas?

Have you been able to find anything else about restarting the backend?

**Reply by mgineer85** [reply]
If it still shows the 2.5.30 there must be something off in the installation. Not sure how you actually installed it. You used pipx?

I applied a patch to the upcoming v5 that should work around the issue if the communication fails as restarts it:
https://github.com/photobooth-app/photobooth-app/issues/385

You could test if if works better for you if you modify the file and change =False to =True instead:
https://github.com/photobooth-app/photobooth-app/commit/07e4e75496746f913f6d9d5077c6a3683f9cee8e#diff-1f70d03f0d711ed7faca0efed38e12c95737a6ff1b12bcded4c7b8522e2d3928L24

Like this:
``` super().__init__(failing_wait_for_lores_image_is_error=True, orientation=config.orientation)```

**Reply by Karl0202** [reply]
Yes, I use "pipx install --system-site-packages photobooth-app --pip-args='--prefer-binary --only-binary gphoto2'", as you suggested.

For my understand, can I test your patch also in my actual release version?

**Reply by mgineer85** [reply]
You can just change the False to True locally in your version, that should have the same effect as the patch.   Don't update to the v5 yet.
